cmd/k8sagent/unit/manifolds.go

```diff
@@ -0,0 +1,330 @@
+		// changes in environment config. We should only need one of
+		// these in a consolidated agent.
+		loggingConfigUpdaterName: ifNotMigrating(wlogger.Manifold(wlogger.ManifoldConfig{
+			AgentName:       agentName,
+			APICallerName:   apiCallerName,
+			LoggingContext:  loggo.DefaultContext(),
+			Logger:          loggo.GetLogger("juju.worker.logger"),
+			UpdateAgentFunc: config.UpdateLoggerConfig,
+		})),
+
+		// The api address updater is a leaf worker that rewrites agent config
+		// as the controller addresses change. We should only need one of
+		// these in a consolidated agent.
+		apiAddressUpdaterName: ifNotMigrating(apiaddressupdater.Manifold(apiaddressupdater.ManifoldConfig{
+			AgentName:     agentName,
+			APICallerName: apiCallerName,
+			Logger:        loggo.GetLogger("juju.worker.apiaddressupdater"),
+		})),
+
+		// The charmdir resource coordinates whether the charm directory is
+		// available or not; after 'start' hook and before 'stop' hook
+		// executes, and not during upgrades.
+		charmDirName: ifNotMigrating(fortress.Manifold()),
+
+		// The leadership tracker attempts to secure and retain leadership of
+		// the unit's service, and is consulted on such matters by the
+		// uniter. As it stands today, we'll need one per unit in a
+		// consolidated agent.
+		leadershipTrackerName: ifNotMigrating(leadership.Manifold(leadership.ManifoldConfig{
+			AgentName:           agentName,
+			APICallerName:       apiCallerName,
+			Clock:               config.Clock,
+			LeadershipGuarantee: config.LeadershipGuarantee,
+		})),
+
+		// HookRetryStrategy uses a retrystrategy worker to get a
+		// retry strategy that will be used by the uniter to run its hooks.
+		hookRetryStrategyName: ifNotMigrating(retrystrategy.Manifold(retrystrategy.ManifoldConfig{
+			AgentName:     agentName,
+			APICallerName: apiCallerName,
+			NewFacade:     retrystrategy.NewFacade,
+			NewWorker:     retrystrategy.NewRetryStrategyWorker,
+			Logger:        loggo.GetLogger("juju.worker.retrystrategy"),
+		})),
+
+		// The uniter installs charms; manages the unit's presence in its
+		// relations; creates subordinate units; runs all the hooks; sends
+		// metrics; etc etc etc. We expect to break it up further in the
+		// coming weeks, and to need one per unit in a consolidated agent
+		// (and probably one for each component broken out).
+		uniterName: ifNotMigrating(uniter.Manifold(uniter.ManifoldConfig{
+			AgentName:             agentName,
+			ModelType:             model.CAAS,
+			APICallerName:         apiCallerName,
+			MachineLock:           config.MachineLock,
+			Clock:                 config.Clock,
+			LeadershipTrackerName: leadershipTrackerName,
+			CharmDirName:          charmDirName,
+			HookRetryStrategyName: hookRetryStrategyName,
+			TranslateResolverErr:  uniter.TranslateFortressErrors,
+			Logger:                loggo.GetLogger("juju.worker.uniter"),
+		})),
+	}
+}
+
+func clockManifold(clock clock.Clock) dependency.Manifold {
```

Comment:
We shouldn't have any more workers that need the clock manifold. We should be passing the clock in as a config param, not getting from another worker.

Pretty sure nothing is using this, so just delete it.